src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/MockHttpMessageHandler.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.AzureDevOpsTimeline.Tests
+{
+    public class MockHttpMessageHandler : HttpMessageHandler
+    {
+        private readonly Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>> _sendAsync;
+
+        public int RequestCancelledCount { get; private set; }
+
+        public MockHttpMessageHandler(Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>> sendAsync)
+        {
+            _sendAsync = sendAsync;
+        }
+
+        public static MockHttpMessageHandler Create(string message)
+        {
+            return new MockHttpMessageHandler(async (request, cancellationToken) =>
+            {
+                var responseMessage = new HttpResponseMessage(HttpStatusCode.OK)
+                {
+                    Content = new StringContent(message)
+                };
+
+                return await Task.FromResult(responseMessage);
+            });
+        }
+
+        public static MockHttpMessageHandler CreateWithCustomHttpStatusCode(HttpStatusCode httpStatusCode)
```

Comment:
Can you maybe just add a second parameter to `Create` with a default set to OK?

Reply:
Yeah, preferably that. Let's not reinvent the wheel

Reply:
With this fake, we are trying to simulate a slow network that we can interrupt with the cancellation token, to see if we are handling that scenario properly. I don't see how we could use the MockHttpClientFactory for this. I'll rename the `MockHttpMessageHandler` to something more descriptive so that it's known that it is very specific